File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    
    static int answer = 0;
        
    public int solution(int[] numbers, int target) {
        
        dfs(0, 0, numbers, target);
        return answer;
    }
    
    public void dfs(int index, int sum, int[] num, int result){
        
        if(index == num.length){
            if(sum == result) answer++;
            return;
        }
        
        dfs(index +1, sum + num[index], num, result);
        dfs(index +1, sum - num[index], num, result);
        
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
var answer = 0;

function solution(numbers, target) {
    
    dfs(0, 0, numbers, target)
    
    return answer
}

function dfs(index, sum, numbers, target){
    
    if(index == numbers.length){
        if(sum == target){
            answer++
        }
        return
    }
    
    dfs(index+1, sum+numbers[index], numbers, target)
    dfs(index+1, sum-numbers[index], numbers, target)
    
}